.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+# Run the job below on every push to master branch.
+on:
+  push:
+    branches:
+      - master
+
+# Runs basic configure, make and make check.
+jobs:
+  tests:
+    name: Build and run the tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Enable source repositories
+        run: |
+          sudo sed -i 's/Types: deb$/Types: deb deb-src/g' \
+                   /etc/apt/sources.list.d/ubuntu.sources
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get -y build-dep libguestfs
+      - name: Install extra dependencies
+        run: |
+          # Git is needed to run git submodule command.
+          # json-c is missing from the Ubuntu package deps.
+          sudo apt-get -y install git libjson-c-dev
+      - name: Fix broken Ubuntu kernel permissions
+        run: |
+          # https://bugs.launchpad.net/ubuntu/+source/linux/+bug/759725
+          sudo chmod 0644 /boot/vmlinuz*
+      - name: Enable KVM
+        run: |
+          sudo chmod 0666 /dev/kvm
```

Comment:
does this work? I did not think github actions support /dev/kvm. well, searching some more their `large runners` support /dev/kvm but those aren't the free ones. https://github.blog/changelog/2023-02-23-hardware-accelerated-android-virtualization-on-actions-windows-and-linux-larger-hosted-runners/

Reply:
Surprisingly yes it does, although I'm not convinced that we're actually using KVM as the tests still run quite slowly.  I think we might as well leave this in since (a) it might work in future and (b) it suppresses this warning:

https://github.com/libguestfs/libguestfs/blob/e7b36e0eba265b0ec44fd782db153e4cac53c5da/lib/launch-direct.c#L149